entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java

```diff
@@ -0,0 +1,39 @@
+package eu.europeana.entitymanagement.solr;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.solr.core.SolrTemplate;
+import org.springframework.data.solr.repository.config.EnableSolrRepositories;
+
+import eu.europeana.entitymanagement.common.config.EntityManagementConfiguration;
+
+@Configuration
+@EnableSolrRepositories
+public class SolrConfig {
```

Comment:
All SolrClient instances (including the two in `ScoringService`) should be initialised here as named beans.

Reply:
I have tried that and I think that Spring has some problems with using many beans with the same type even though I tried to distinguish them by using the bean names and the @Qualifier annotation while imorting them over @Autowired. Therefore I actually created the indexing SolrClient the same way as the other 2 solr clients in the ScoringService class.

Reply:
See [this commit](https://github.com/europeana/entity-management/compare/EA-2577-solr-indexing...EA-2577-solr-config).

Reply:
I see, the problem was in the SolrTemplate beans that did not work together with the SolrClient beans and you put the @Qualifier in the constructor. Fine.